### PR TITLE
Update dependencies to address CVEs

### PR DESCRIPTION
**Pull request by AlexsandroRyan**
This pull request addresses all CVEs reported by the Checkmarx tool during its execution on this repository.

The previous discussion can be found here: https://github.com/cometbft/cometbft/discussions/3558.

I have updated all necessary dependencies to fix the identified CVEs, but some vulnerabilities remain unresolved. I would appreciate any assistance in addressing these remaining issues.

1. CVE-2021-3538: This issue is related to the github.com/satori/go.uuid package, which is a dependency of [tm-load-test](https://github.com/informalsystems/tm-load-test). We have already submitted a PR to address this: https://github.com/informalsystems/tm-load-test/pull/221.

2. CVE-2024-24786: This vulnerability pertains to the google.golang.org/protobuf package. Running go mod graph | grep google.golang.org/protobuf reveals that many packages are using the vulnerable version. It’s unclear if updating them individually is feasible.

3. CVE-2024-34478: This vulnerability is associated with github.com/btcsuite/btcd, a dependency of github.com/btcsuite/btcd/btcutil, which is currently used at a version lower than 0.24.0. We have also submitted a pull request for this: https://github.com/btcsuite/btcd/pull/2235.

Please let us know if this approach is sufficient or if there is a more efficient way to resolve these issues.

---

#### PR checklist

- [ ] ~Tests written/updated~
- [ ] ~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~
- [ ] ~Updated relevant documentation (`docs/` or `spec/`) and code comments~
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,81 +1,89 @@
 module github.com/tendermint/tendermint
 
-go 1.21
+go 1.21.0
```
Comment (melekes) [inline go.mod:3]:
```suggestion
go 1.21.13
```

why not the latest minor release?

**Review by melekes — APPROVED** [review]
👍

File: go.mod (hunk 1 above)
Comment (AlexsandroRyan) [reply go.mod]: @melekes Oh, probably missed that, just updated :+1:

**Comment by melekes** [comment]
Closing in favor of https://github.com/cometbft/cometbft/pull/3754 because we can't push to your fork (you need to allow maintainers to push to your PR in the GH settings).

**Pull request closed without merging**